src/primitives.jl

```diff
@@ -325,14 +359,34 @@ end
     _mapreducetasks(applyreduce_array, op, eachindex(A), threaded; init)
 end
 # Try to applyreduce over iterables
-@inline function _applyreduce(f::F, op::O, target, ::Nothing, iterable; threaded, init) where {F, O}
-    applyreduce_iterable(i) = _applyreduce(f, op, target, x; threaded=_False(), init)
-    if threaded # Try to `collect` and reduce over the vector with threads
-        _applyreduce(f, op, target, collect(iterable); threaded, init)
+@inline function _applyreduce(f::F, op::O, target, ::Nothing, iterable::IterableType; threaded, init) where {F, O, IterableType}
+    if Tables.istable(iterable)
+        _applyreduce_to_table(f, op, target, iterable; threaded, init)
     else
-        # Try to `mapreduce` the iterable as-is
-        mapreduce(applyreduce_iterable, op, iterable; init)
+        applyreduce_iterable(i) = _applyreduce(f, op, target, x; threaded=_False(), init)
+        if threaded isa _True # Try to `collect` and reduce over the vector with threads
+            _applyreduce(f, op, target, collect(iterable); threaded, init)
+        else
+            # Try to `mapreduce` the iterable as-is
+            mapreduce(applyreduce_iterable, op, iterable; init)
+        end
+    end
+end
+# In this case, we don't reconstruct the table, but only operate on the geometry column.
+function _applyreduce_to_table(f::F, op::O, target, iterable::IterableType; threaded, init) where {F, O, IterableType}
```

Comment:
I'm not sure it should have `to` in the function name as it doesn't return a table?